src/toast/tests/instrument.py

```diff
@@ -87,12 +96,22 @@ def test_focalplane_full(self):
         fp_file = os.path.join(self.outdir, "focalplane_full.h5")
         check_file = os.path.join(self.outdir, "check_full.h5")
 
-        if self.comm is None or self.comm.rank == 0:
-            fp.save_hdf5(fp_file)
+        hf = hdf5_open(fp_file, "w", comm=self.comm)
+        fp.save_hdf5(hf, comm=self.comm)
+        if hf is not None:
+            hf.close()
+        del hf
+
         if self.comm is not None:
             self.comm.barrier()
 
-        newfp = Focalplane(file=fp_file, comm=self.comm)
+        newfp = Focalplane()
```

Comment:
This new interface inflates the amount of code needed to load a focalplane in the workflow from 1 line to 6 lines. Sometimes that is unavoidable but could that complexity be hidden in the `Focalplane` class?

Reply:
The latest commit introduces an `H5File` wrapper class that acts as a context manager for opening HDF5 files in parallel or serial or on one process of a communicator.  This still expands 1 line into 3 lines, but hopefully makes it easier for high-level code to open a file and load simple objects (like a focalplane or noise model) without having to add duplicated file handling features to the focalplane and noise classes (and soon the Telescope, Site, Weather, etc classes).